worker/awsCertlint/awsCertlint.go

```diff
@@ -0,0 +1,167 @@
+package awsCertlint
+
+import (
+	"bufio"
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"strings"
+
+	"github.com/mozilla/tls-observatory/certificate"
+	"github.com/mozilla/tls-observatory/worker"
+)
+
+var (
+	workerName = "awsCertlint"
+	workerDesc = "Runs awslabs/certlint over a given Certificate, categoriezes output for display on the certificate"
+
+	certlintDirectory = "/go/certlint" // path from tools/Dockerfile-scanner
+)
+
+type Result struct {
+	Bugs []string `json:"Bugs"`
+	Informational []string `json:"Informational"`
+	Notices []string `json:"Notices"`
+	Warnings []string `json:"Warnings"`
+	Errors []string `json:"Errors"`
+	FatalErrors []string `json:"FatalErrors"`
+}
+
+func init() {
+	runner := new(eval)
+	worker.RegisterWorker(workerName, worker.Info{Runner: runner, Description: workerDesc})
+}
+
+type eval struct{}
+
+func (e eval) Run(in worker.Input, resChan chan worker.Result) {
+	result := worker.Result{
+		WorkerName: workerName,
+	}
+	lintResult, err := e.runCertlint(in.Certificate)
+	if err != nil {
+		resChan <- worker.Result{
+			Success:    false,
+			WorkerName: workerName,
+			Errors:     []string{fmt.Sprintf("%s for certificate %s", err.Error(), in.Certificate.ID)},
+			Result:     nil,
+		}
+		return
+	}
+
+	// Marshal the response
+	bs, err := json.Marshal(&lintResult)
+	if err != nil {
+		result.Success = false
+		result.Errors = append(result.Errors, err.Error())
+	} else {
+		result.Success = true
+		result.Result = bs
+	}
+	resChan <- result
+}
+
+func (e eval) runCertlint(cert certificate.Certificate) (*Result, error) {
+	tmp, err := ioutil.TempFile("", "awslabs-certlint")
+	if err != nil {
+		return nil, fmt.Errorf("error creating temp dir", cert.ID)
+	}
+	defer os.Remove(tmp.Name())
+	x509Cert, err := cert.ToX509()
+	if err != nil {
+		return nil, fmt.Errorf("error converting to x509.Certificate", cert.ID)
+	}
+	if err := ioutil.WriteFile(tmp.Name(), x509Cert.Raw, 0644); err != nil {
+		return nil, fmt.Errorf("error writing x509.Certificate to temp file, err=%v", err)
+	}
+
+	// Run certlint over certificate
+	cmd := exec.Command("ruby", "-I", "lib:ext", "bin/certlint", tmp.Name())
```

Comment:
I like to run those with a timeout to prevent them from hanging. Take a look at https://github.com/mozilla/tls-observatory/blob/master/connection/retriever.go#L47-L56